src/core/patch.rkt

```diff
@@ -131,30 +134,41 @@
   (define reprs (map (curryr repr-of (*context*)) exprs))
   (timeline-push! 'inputs (map ~a exprs))
   (define runner (make-egg-runner exprs reprs schedule #:context (*context*)))
-  (define variantss (run-egg runner `(multi . ,extractor)))
+  ; variantss is a (listof roots))
+  (define rootss (run-egg runner `(multi . ,extractor)))
 
   ; apply changelists
   (define rewritten
     (reap [sow]
-          (for ([variants (in-list variantss)]
+          (for ([roots (in-list rootss)]
                 [altn (in-list altns)])
-            (for ([variant (in-list (remove-duplicates variants))])
-              (sow (alt variant (list 'rr runner #f #f) (list altn) '()))))))
+            (for ([root (in-list (remove-duplicates roots))])
+              (sow (alt (batchref global-batch root) (list 'rr runner #f #f) (list altn) '()))))))
```

Comment:
This is very cool, though maybe it makes more sense for `run-egg` to return the `batchref` objects directly?

Reply:
Done